core/modules/replica/IngestHttpSvcMod.cc

```diff
@@ -155,19 +167,91 @@ json IngestHttpSvcMod::_readLocal(string const& filename) {
+    } catch (DatabaseServicesNotFound const&) {}
+
+    string caPath;
+    string caInfo;
+    string caInfoTmp;
+    if (sslVerifyPeer) {
+        // Get a path to a folder with host certificates, if any was configured
+        // for this catalog.
+        try {
+            caPath = databaseServices->ingestParam(
+                    database,
+                    HttpFileReaderConfig::category,
+                    HttpFileReaderConfig::caPathKey).value;
+        } catch (DatabaseServicesNotFound const&) {}
+
+        // Get a path to an existing file with peer certificate, if any was configured
+        // for this catalog.
+        try {
+            caInfo = databaseServices->ingestParam(
+                    database,
+                    HttpFileReaderConfig::category,
+                    HttpFileReaderConfig::caInfoKey).value;
+        } catch (DatabaseServicesNotFound const&) {}
+
+        // The certificate bundle (if any) loaded into the Ingest configuration by value
+        // will superseed the one set above.
+        try {
+            auto const caInfoVal = databaseServices->ingestParam(
+                    database,
+                    HttpFileReaderConfig::category,
+                    HttpFileReaderConfig::caInfoValKey).value;
+            if (!caInfoVal.empty()) {
+                // Write certificate bundle into a file created at the temporary folder
+                // configured to support HTTP-based file ingest operations. The file get
+                // destroyed after finishing the ingest.
+                string const baseDir = workerInfo().httpLoaderTmpDir;
+                string const prefix = database + "-";
+                string const model = "%%%%-%%%%-%%%%-%%%%";
+                string const suffix = ".cert";
+                unsigned int const maxRetries = 1;
+                caInfoTmp = FileUtils::createTemporaryFile(baseDir, prefix, model, suffix, maxRetries);
+                ofstream fs;
+                fs.open(caInfoTmp, ios::out|ios::trunc);
+                if (!fs.is_open()) {
+                    HttpError(__func__, "failed to open/create file '" + caInfoTmp+ "'.");
+                }
+                fs << caInfoVal;
+                fs.flush();
+                fs.close();
+                // Use this file instead of the existing path.
+                caInfo = caInfoTmp;
+            }
+        } catch (DatabaseServicesNotFound const&) {}
+    }
+
+    // Read the file from the data source
+    HttpFileReader reader(method, url, data, headers, sslVerifyPeer, caPath, caInfo);
+    reader.read([this, &numBytes, &numRows](string const& row) {
         this->writeRowIntoFile(row);
         numBytes += row.size() + 1;     // counting the newline character
         ++numRows;
     });
+    if (!caInfoTmp.empty()) {
+        // Make the best effort to delete the file. Ignore any errors.
+        boost::system::error_code ec;
+        fs::remove(fs::path(caInfoTmp), ec);
```

Comment:
IF exception happens in `reader.read()` then file is not removed? I'd move all code that creates/removes temporary file into a separate class and use RAII to manage its lifetime.

Reply:
Good suggestion! I too thought about it. Though, there is nothing wrong in leaving those weirdly (uniquely) named tiny files in the temporary filesystem. Then file system could be cleaned up when the server would be starting next time.